### PR TITLE
WIP Cherry pick for legacy

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
### DO NOT SQUASH

Waiting on #881

### CODE REVIEW

File: src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs [hunk 1]
```diff
@@ -1231,6 +1234,54 @@ public static string GetDecoratedSymbolName(SymbolReference symbolReference)
...
+
+                var psCommand = new PSCommand();
+
+                if (isNotCurrentProcess) {
+                    psCommand.AddCommand("Enter-PSHostProcess").AddParameter("Id", processId).AddStatement();
+                }
+
+                psCommand.AddCommand("Get-Runspace");
+
+                StringBuilder sb = new StringBuilder();
```
Comment (rkeithhill) [inline src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs]: What happened to the `var` here?

File: src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs [hunk 2]
```diff
@@ -1231,6 +1234,54 @@ public static string GetDecoratedSymbolName(SymbolReference symbolReference)
...
+        {
+            var runspaceResponses = new List<GetRunspaceResponse>();
+
+            if (this.editorSession.PowerShellContext.LocalPowerShellVersion.Version.Major >= 5)
+            {
+                if (processId == null) {
+                    processId = "current";
+                }
+
+                var isNotCurrentProcess = processId != null && processId != "current";
```
Comment (rkeithhill) [inline src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs]: A few parens here would make this a bit easier to visually parse.

File: src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs (hunk 1 above)
Comment (TylerLeonhardt) [reply src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs]: That's in #881

File: src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs (hunk 2 above)
Comment (TylerLeonhardt) [reply src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs]: Also fixed in #881

**Comment by TylerLeonhardt** [comment]
not sure why appveyor isn't triggering a build...

**Comment by TylerLeonhardt** [comment]
since the implementation of #881 would need to be different for legacy, I'm going to merge this in and then send a PR to legacy with that implementation.

**Pull request merged**